app/src/main/java/org/wikipedia/analytics/eventplatform/EventPlatformClientIntegration.java

```diff
@@ -0,0 +1,143 @@
+package org.wikipedia.analytics.eventplatform;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import org.wikipedia.WikipediaApp;
+import org.wikipedia.dataclient.ServiceFactory;
+import org.wikipedia.dataclient.WikiSite;
+import org.wikipedia.settings.Prefs;
+import org.wikipedia.util.log.L;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
+import io.reactivex.rxjava3.disposables.CompositeDisposable;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+import static java.net.HttpURLConnection.HTTP_ACCEPTED;
+import static java.net.HttpURLConnection.HTTP_BAD_REQUEST;
+import static java.net.HttpURLConnection.HTTP_CREATED;
+import static java.net.HttpURLConnection.HTTP_GATEWAY_TIMEOUT;
+import static java.net.HttpURLConnection.HTTP_INTERNAL_ERROR;
+import static java.net.HttpURLConnection.HTTP_UNAVAILABLE;
+import static org.wikipedia.BuildConfig.META_WIKI_BASE_URI;
+import static org.wikipedia.settings.Prefs.getEventPlatformSessionId;
+import static org.wikipedia.settings.Prefs.getStreamConfigs;
+import static org.wikipedia.settings.Prefs.setEventPlatformSessionId;
+import static org.wikipedia.settings.Prefs.setStreamConfigs;
+import static org.wikipedia.util.DateUtil.iso8601DateFormat;
+
+final class EventPlatformClientIntegration {
+
+    private static final Map<String, EventService> RETROFIT_SERVICE_CACHE = new HashMap<>();
+    private static final CompositeDisposable DISPOSABLES = new CompositeDisposable();
```

Comment:
This list of disposables is "added" to, but it's never "cleared".  This list will keep growing without bounds, and eventually run out of memory.

Reply:
It'd be nice to briefly document what's happening here and why, it would help orient engineers who may be less familiar with the app's conventions.

Reply:
I have cleared it after use. @dbrant lmk if there is a better way of doing it.